### PR TITLE
Webapp actions/selectors for channels

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
More webapp actions/selectors. @enahum note that I changed the `members` channel entity to be `membersInChannel` and follow the same pattern as the one for teams.

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by enahum** [comment]
@jwilander this is failing, now another question would be why the in the entities reducer?

File: src/actions/channels.js [hunk 1]
```diff
@@ -117,6 +117,10 @@ export function createDirectChannel(userId, otherUserId) {
             last_update_at: created.create_at
         };
 
+        const preference = [{category: Preferences.CATEGORY_DIRECT_CHANNEL_SHOW, name: otherUserId, value: 'true'}];
+
+        savePreferences(userId, preference);
```
Comment (enahum) [inline src/actions/channels.js]: note to self: remove the save preference in the RN action when creating a direct channel as is going to be handled by this

File: src/actions/channels.js [hunk 2]
```diff
@@ -128,7 +132,7 @@ export function createDirectChannel(userId, otherUserId) {
             },
             {
                 type: PreferenceTypes.RECEIVED_PREFERENCES,
-                data: [{category: Preferences.CATEGORY_DIRECT_CHANNEL_SHOW, name: otherUserId, value: 'true'}]
+                data: preference
```
Comment (enahum) [inline src/actions/channels.js]: can we name this `preferences` as is an array and not just the preference object?

**Comment by jwilander** [comment]
It passed, locally...

You mean, why the `membersInChannel`? Because it's an object with channel id keys of objects with user id keys with members as values

File: src/reducers/entities/channels.js [hunk 3]
```diff
@@ -181,33 +179,47 @@ function myMembers(state = {}, action) {
     }
 }
 
-function members(state = {}, action) {
-    const nextState = {...state};
-
+function membersInChannel(state = {}, action) {
```
Comment (enahum) [inline src/reducers/entities/channels.js]: note to self: Rename `members` to `membersInChannel` wherever this is used in RN

File: src/selectors/entities/channels.js [hunk 4]
```diff
@@ -28,8 +32,8 @@ export function getMyChannelMemberships(state) {
     return state.entities.channels.myMembers;
 }
 
-export function getChannelMembers(state) {
-    return state.entities.channels.members;
+export function getChannelMembersInChannels(state) {
```
Comment (enahum) [inline src/selectors/entities/channels.js]: note to self: rename this too in the RN app

File: src/actions/channels.js (hunk 2 above)
Comment (jwilander) [inline src/actions/channels.js]: Yeah good point, I'll update this

**Comment by jwilander** [comment]
Merging because I'm pretty sure the random test failures aren't related to this PR

**Pull request merged**